### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by batoorsohail**
### In this PR we:

- [x] Created a model for users, posts, comments, and users
- [x] Created a method that returns the 3 most recent posts for a given user.
- [x] Created a method that updates the posts counter for a user.
- [x] Created a method which returns the 5 most recent comments for a given post.
- [x] Created a method that updates the comments counter for a post.
- [x] Created a method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_author_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (julie-ify) [inline app/models/models/post.rb]: - [ ] **[OPTIONAL]** I would recommend you make this method private and attach an ActiveRecord call back to it.

- [ ] **[OPTIONAL]** I also think we can use `increment!` method here.

  ```
  after_create :update_author_posts_counter

  ...

  private

  def update_author_posts_counter
    author.increment!(:posts_counter)
  end
  ```

File: app/models/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (julie-ify) [inline app/models/models/like.rb]: - [ ] **[OPTIONAL]** I would recommend you make this method private and attach an ActiveRecord call back to it.

- [ ] **[OPTIONAL]** I also think we can use `increment!` method here.

  ```
  after_create :update_post_likes_counter

  ...

  private

  def update_post_likes_counter
    post.increment!(:likes_counter)
  end
  ```

File: app/models/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (julie-ify) [inline app/models/models/comment.rb]: - [ ] **[OPTIONAL]** You can also attach a callback that triggers this method automatically per previous suggestions.

**Review by julie-ify — APPROVED** [review]
Hi @batoorsohail,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by batoorsohail** [comment]
Thanks @julie-ify 😄

**Pull request merged**